### PR TITLE
Add pytest CI to run tests on every push (fix #26)

### LINKED ISSUE #26: Intergrate CI to run pytest on every  merged PR
# Situation

On every  merged PR, we need to run our tests to check the functionality  of our tool.

# Proposed Solution 

Use Github Actions to automate this process.

**Subtasks** 

* [ ] Create a Github Actions workflow 
* [ ] Add coverage

### PR DESCRIPTION
**Pull request by SchleichsSalaticus**
Add a CI badge to README
Add a CI workflow to run pytest

### CODE REVIEW

File: .github/workflows/pytest.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}    
+
+    - name: Test python version [3.6, 3.7, 3.8, 3.9] and install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install pytest pytest-cov
```
Comment (tomschr) [inline .github/workflows/pytest.yml:30]: If you hardcode the dependencies, you need to change the file all the time. Better try indirection and use the requirement file here:

```suggestion
        pip install -r devel-requirements.txt
```

File: .github/workflows/pytest.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0 # otherwise, you will failed to push refs to dest repo
+
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}    
+
+    - name: Test python version [3.6, 3.7, 3.8, 3.9] and install dependencies
```
Comment (tomschr) [inline .github/workflows/pytest.yml:27]: Not sure, but I think you can use a placeholder here:

```suggestion
    - name: Test python version ${{ matrix.python-version }} and install dependencies
```

File: .github/workflows/pytest.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      run: |
+        python -m pip install --upgrade pip
+        pip install pytest pytest-cov
+
+    - name: Run Tests and create coverage
+      run: |
+        pytest -v --cov=reportdaily tests/
```
Comment (tomschr) [inline .github/workflows/pytest.yml:34]: I'd recommend to move all options into the `setup.cfg` file and call just `pytest` or `pytest -v`.  That way you can adapt the config file without changing this file.

**Review by tomschr — CHANGES_REQUESTED** [review]
@SchleichsSalaticus This looks very good! :+1: Just a few minor details which would make changes in the future easier.

File: .github/workflows/pytest.yml (hunk 1 above)
Comment (SchleichsSalaticus) [reply .github/workflows/pytest.yml:30]: Thanks this makes totally sence :D

**Pull request merged**